src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -73,8 +73,12 @@ symbolt generate_java_generic_typet::operator()(
     pre_modification_size==after_modification_size,
     "All components in the original class should be in the new class");
 
-  const java_specialized_generic_class_typet new_java_class =
-    construct_specialised_generic_type(new_tag, replacement_components);
+  const java_specialized_generic_class_typet new_java_class{
+    new_tag,
```

Comment:
I suggest we call this `generic_tag` rather than `new_tag` to emphasize the difference.

Reply:
Good spot - perhaps `generic_name` since not a tag at all?

Reply:
True, in that case it would be good to rename the function too - from `build_generic_tag` to `build_generic_name`.